documentation/src/docs/controller-service.md

```diff
@@ -282,168 +277,193 @@ Controller service efficient queries for finding immediate successors
  class which allows and prohibits various state transitions.
- Stream state describes the current state of the stream. It transitions
+ Stream State describes the _current state_ of the Stream. It transitions
  from _active_ to respective action based on the action being performed
- on the stream. For example, during scaling the state of the stream
+ on the Stream. For example, during scaling the state of the Stream
  transitions from *active* to *scaling* and once scaling completes, it
- transitions back to *active*. Stream state is used as a barrier to
- ensure only one type of operation is being performed on a given stream
+ transitions back to *active*. Stream State is used as a barrier to
+ ensure only one type of operation is being performed on a given Stream
  at any point in time. Only certain state transitions are allowed and
- are described in the State Transition object. Only legitimate state
+ are described in the state transition object. Only legitimate state
  transitions are allowed and any attempt for disallowed transition
  results in appropriate exception.
 
 #### Truncation Record
- This corresponds to the stream cut which was last used to truncate the
- given stream. All stream segment queries superimpose the truncation
- record and return segments that are strictly greater than or equal to
- the stream cut in truncation record.
+ This corresponds to the `StreamCut` which was last used to truncate the
+ given Stream. All Stream Segment queries superimpose the truncation
+ record and return Stream Segments that are strictly greater than or equal to
+ the `StreamCut` in truncation record.
+
+#### Sealed Segments Maps
+Once the Stream Segments are sealed, controller needs to store additional information about the Stream Segment. Presently, we have two types of information:
+
+ - Epoch, the Stream Segment was sealed in.
+ - Size of the Stream Segment at the time of sealing.
+
+ These records have two different characteristics and are used in different types of queries. For example;
+
+ 1. Sealing epoch is important for querying _successor Stream Segments_.(For each Stream Segment we store its sealing epoch directly in the metadata store.)
+ 2. Stream Segment sizes are used during truncation workflows.(For sealed sizes, we store it in a map of Segment to size at the time of sealing.)
+ 3. Successor queries are performed on a single Stream Segment whereas truncation workflows work on a group of Stream Segments.
+
+ This ensures that during truncation we are able to retrieve sealed sizes for multiple Stream Segments with minimal number of calls into underlying metadata store. Since we could have arbitrarily large number of Stream Segments that have been sealed away, we cannot store all of the information in a single map and hence we shard the map and store it. The sharding function we use is to hash the creation epoch and get the shard number.
```

Comment:
@shivesh Sealed Sizes?? can u rephrase?

Reply:
sizes of segments at the time of their respective sealing

Reply:
Accepted suggestion